src/extension/background-script/connectors/commando.ts

```diff
@@ -0,0 +1,334 @@
+    await this.ln.connect();
+  }
+
+  async unload() {
+    await this.ln.disconnect();
+  }
+
+  async connectPeer(
+    args: ConnectPeerArgs
+  ): Promise<ConnectPeerResponse | Error> {
+    return this.ln
+      .commando({
+        method: "connect",
+        params: {
+          id: args.pubkey,
+          host: args.host,
+        },
+        rune: this.config.rune,
+      })
+      .then((resp) => {
+        return {
+          data: true,
+        };
+      })
+      .catch((err) => {
+        return new Error(err);
+      });
+  }
+
+  async getInvoices(): Promise<GetInvoicesResponse> {
+    return this.ln
+      .commando({
+        method: "listinvoices",
+        params: {},
+        rune: this.config.rune,
+      })
+      .then((resp) => {
+        const parsed = resp as CommandoListInvoicesResponse;
+        return {
+          data: {
+            invoices: parsed.invoices.map(
+              (invoice, index): ConnectorInvoice => ({
+                id: invoice.label,
+                memo: invoice.description,
+                settled: invoice.status == "paid",
+                preimage: invoice.payment_preimage,
+                settleDate: invoice.paid_at,
+                type: "received",
+                totalAmount: (invoice.amount_received_msat / 1000).toString(),
+              })
+            ),
+          },
+        };
+      });
+  }
+
+  async getInfo(): Promise<GetInfoResponse> {
+    const response = (await this.ln.commando({
+      method: "getinfo",
+      params: {},
+      rune: this.config.rune,
+    })) as CommandoGetInfoResponse;
+    return {
+      data: {
+        alias: response.alias,
+        pubkey: response.id,
+        color: response.color,
+      },
+    };
+  }
+
+  async getBalance(): Promise<GetBalanceResponse> {
+    const response = (await this.ln.commando({
+      method: "listfunds",
+      params: {},
+      rune: this.config.rune,
+    })) as CommandoListFundsResponse;
+    let lnBalance = 0;
+    for (let i = 0; i < response.channels.length; i++) {
+      lnBalance = lnBalance + response.channels[i].channel_sat;
+    }
+    return {
+      data: {
+        balance: lnBalance,
+      },
+    };
+  }
+
+  async sendPayment(args: SendPaymentArgs): Promise<SendPaymentResponse> {
+    return this.ln
+      .commando({
+        method: "pay",
+        params: {
+          bolt11: args.paymentRequest,
+        },
+        rune: this.config.rune,
+      })
+      .then((resp) => {
+        const parsed = resp as CommandoPayInvoiceResponse;
+        const parsedTotalAmtStr = parsed.amount_msat.replace("msat", "");
+        const parsedSentAmtPaidStr = parsed.amount_sent_msat.replace(
+          "msat",
+          ""
+        );
+        const parsedTotalAmt = (parsedTotalAmtStr as unknown as number) / 1000;
+        const parsedTotalFee =
+          ((parsedSentAmtPaidStr as unknown as number) -
+            (parsedTotalAmtStr as unknown as number)) /
```

Comment:
I think @escapedcat should look at this TypeScript

Reply:
Updated